src/lib/tools.ts

```diff
@@ -0,0 +1,50 @@
+import { SerpAPI } from 'langchain/tools';
+import { Calculator } from 'langchain/tools/calculator';
+import { DynamicTool, DynamicStructuredTool } from 'langchain/tools';
+import { QdrantClient } from '@qdrant/js-client-rest';
+import { OpenAIEmbeddings } from 'langchain/embeddings/openai';
+import { z } from 'zod';
+
+function getTools() {
+	if (import.meta.env.VITE_SERPAPI_API_KEY == undefined) {
+		import.meta.env.VITE_SERPAPI_API_KEY = 'no_key';
+		throw console.warn('SERPAPI_API_KEY is undefined');
+	}
+	const tools = [
+		new DynamicTool({
+			//for testing remove when merge
+			name: 'FRC971Database',
+			description: 'useful to get information about FRC971. Input should be a search query.',
+			func: async (query: string) => {
+				const embeddings = new OpenAIEmbeddings({ openAIApiKey: process.env.OPENAI_API_KEY });
+				const client = new QdrantClient({ host: 'localhost', port: 6333 });
```

Comment:
I'd prefer for ChatCompletion to hold a QdrantClient and for it to pass it into the getTools function so we don't create a new one every time we get our tools.

Reply:
It would be hard to pass in the qdrant client into the function because I would need to create a custom agent execute tool function. However, we can set embeddings and client as global variables. What do you think?

Reply:
Nvm, I think I found a solution